gramps/plugins/view/relview.py

```diff
@@ -1107,9 +1107,33 @@ def write_parents(self, family_handle, person=None):
             )
             self.row += 1  # now advance it
         else:
-            self.write_label(_("%s:") % _("Parents"), family, True, person)
-            self.write_person(_("Father"), family.get_father_handle())
-            self.write_person(_("Mother"), family.get_mother_handle())
+            if family:
+                father = mother = None
+                hd21 = family.get_father_handle()
+                if hd21:
+                    father = self.dbstate.db.get_person_from_handle(hd21).gender
+                hd22 = family.get_mother_handle()
+                if hd22:
+                    mother = self.dbstate.db.get_person_from_handle(hd22).gender
+
+                parent1 = parent2 = ""
+                if father == Person.MALE:
+                    parent1 = _("Father")
+                elif father == Person.FEMALE:
+                    parent1 = _("Mother")
+                else:
+                    parent1 = _("Parent")
```

Comment:
Consider moving this block into a function to avoid duplication.
Something like this:
```
parent1 = self.__get_parent_label(father)
parent2 = self.__get_parent_label(mother)
```

Reply:
Admittedly, I am not a coder. I can hack what others have thought up.

The process I have is to determine the label for the person slotted in the Father/Parent1 slot based upon their gender and then do the same for the person in the Mather/Parent2 slot.

Using my code, I actually have a Female in the F/P1 slot and the father/sperm donor in the M/P2 slot.\ for the labels Mother and Father.